_posts/2023-08-29-kafka-node-pools-supporting-kraft.md

```diff
@@ -0,0 +1,178 @@
+Then you have the Kafka cluster with one or more Apache Kafka brokers that is bootstrapped using the ZooKeeper cluster.
+And that's it.
+There aren't really any other options
+But when we remove ZooKeeper to introduce the KRaft (Kafka Raft metadata) mode, the architectural landscape changes significantly.
+In this blog post, we will look at the various KRaft architectures and how Strimzi can support them using the Kafka node pools.
+
+<!--more-->
+
+_This post is part of a series about Kafka node pools.
+Other posts include:_
+
+* _[Part 1 - Introduction](https://strimzi.io/blog/2023/08/14/kafka-node-pools-introduction/)_
+* _[Part 2 - Node ID Management](https://strimzi.io/blog/2023/08/23/kafka-node-pools-node-id-management/)_
+* _[Part 3 - Storage & Scheduling](https://strimzi.io/blog/2023/08/28/kafka-node-pools-storage-and-scheduling/)_
+* _Part 4 - Supporting KRaft (this post)_
+
+### KRaft architectures
+
+In the KRaft mode, the Kafka cluster runs without an Apache ZooKeeper cluster as a dependency.
+The Kafka nodes in the KRaft mode have _controller_ and/or _broker_ roles.
+The nodes with the _controller_ role are responsible for maintaining the Kafka metadata, handling leader elections, and helping to bootstrap the Kafka cluster.
+They assume the responsibilities previously handled by the ZooKeeper nodes.
+The nodes with the _broker_ role have the same responsibilities as the Kafka brokers in the ZooKeeper-based Kafka cluster.
+They receive the messages from the producers, store them, and pass them to the consumers.
+One node can also have both roles - a controller and a broker -  at the same time.
+This offers a number of configuration options for your KRaft-based Kafka cluster.
+
+You might start with a small Kafka cluster consisting only of combined nodes that all have controller and broker roles.
+Typically, it would have three nodes.
+But if needed you can also run _a whole Kafka cluster_ in a single node.
+Such a cluster is suitable for example for development and testing.
+It might be also useful in other resource-constraint environments.
+For example when running Kafka at the edge.
+
+![Three-node Kafka cluster with combined nodes](/assets/images/posts/2023-08-29-kafka-node-pools-supporting-kraft-combined-nodes.png)
+
+For a production cluster, you would normally use dedicated controller nodes and dedicated broker nodes.
+You would typically maintain three controller nodes for availability, and add as many broker nodes as you need.
+This is also the architecture to which you would migrate from a ZooKeeper-based cluster.
+
+![Three-node Kafka cluster with dedicated controller and broker nodes](/assets/images/posts/2023-08-29-kafka-node-pools-supporting-kraft-separate-nodes.png)
+
+And somewhere in between is an architecture where some nodes have both controller and broker roles and some have only the broker role.
+It is also an intermediate step when migrating between the previous two architectures. 
```

Comment:
Let's take in mind that migration from ZooKeeper-based cluster to a KRaft one with nodes in "combined mode" is not possible. Not sure if this sentence could be misleading.

Reply:
Why do you think it would be misleading? If you want to migrate from dedicated controller nodes to combined controller nodes (or the other way around), you would likely have to transition through this stage.

Reply:
So, should I instead use something like this?

```
It is also an intermediate step when migrating between the combined nodes to the dedicated controller nodes or the other way around. 
```

Reply:
Yeah I would say this is a good one! Thanks!